File: 0383-ransom-note/0383-ransom-note.java
import java.util.Arrays;

class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
       char[] s = ransomNote.toCharArray();
        char[] m =magazine.toCharArray();
        int[] res =new int[s.length];
        int flag = 0;
        for (int i=0;i<s.length;i++) {
            for(int j=0;j<m.length;j++){
                if(m[j] ==s[i]){
                    res[i] =1;
                    m[j]='9';
                    break;
                }
                
            }
        }
        flag = Arrays.stream(res).min().getAsInt();

        return flag==1;
    }
}

File: 0383-ransom-note/0383-ransom-note.cs
public class Solution {
    public bool CanConstruct(string ransomNote, string magazine) {
        int[] freq  = new int[26];

        
        foreach(var e in magazine){
            freq[e-'a']++;
        }
        foreach(var e in ransomNote){
            freq[e-'a']--;
        }
        foreach(var e in freq){
            if(e<0){
                return false;
            }
        }
        return true;
    }
}